### PR TITLE
[Issue-1617] - Traffic Ops Keeps track of configuration differences between database and Traffic Servers

### PR DESCRIPTION
**Pull request by KevinMackenzie**
PR for feature #1617

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: traffic_ops/app/db/create_tables.sql [hunk 1]
```diff
@@ -2596,6 +2595,14 @@ ALTER TABLE ONLY asn
     ADD CONSTRAINT fk_cran_cachegroup1 FOREIGN KEY (cachegroup) REFERENCES cachegroup(id);
 
 
+--
+-- Name: server; Type: DEFAULT; Schema: public; Owner: traffic_ops
+--
+
+ALTER TABLE ONLY config_diffs
```
Comment (mitchell852) [inline traffic_ops/app/db/create_tables.sql]: this should be in your migration file. no need to touch this file.

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by limited** [comment]
ok to test

**Comment by limited** [comment]
ok to test

**Comment by dangogh** [comment]
@KevinMackenzie this PR has a lot of merge commits,  which makes it difficult to examine.   Can you use `git rebase -i master` and get rid of those?

**Comment by KevinMackenzie** [comment]
I rebased my fork to remove all of the merged commits and then merged the `routes.go` as the last item.  Is  this OK?

**Comment by limited** [comment]
@KevinMackenzie Github still shows conflicts in the PR. Can you please take another look?

**Pull request closed without merging**